docs/source/mypy_daemon.rst

```diff
@@ -0,0 +1,88 @@
+.. _mypy_daemon:
+
+Mypy daemon (mypy server)
+=========================
+
+Instead of running mypy as a command-line tool, you can also run it as
+a long-running daemon (server) process and use a command-line client to
+control the server.  This way mypy can perform type checking much faster,
+since program state cached from previous runs is kept in memory and doesn't
+have to be read from the file system on each run. The server also uses
+finer-grained dependency tracking to reduce the amount of work that needs
+to be done.
+
+If you have a large codebase to check, running mypy using the mypy
+daemon can be *10 or more times faster* than the regular command-line
+``mypy`` tool, especially if your workflow involves running mypy
+repeatedly after small edits -- which is often a good idea, as this way
+you'll find errors sooner.
+
+.. note::
+
+    The mypy daemon is experimental. In particular, the command-line
+    interface may change in future mypy releases.
+
+.. note::
+
+    The mypy daemon currently supports macOS and Linux only.
+
+Basic usage
+***********
+
+The client utility ``dmypy`` is used to control the mypy daemon.
+Use ``dmypy start -- <flags>`` to start the daemon. You can use almost
+arbitrary mypy flags after ``--``.  The daemon will always run on the
+current host. Example::
+
+    dmypy start -- --follow-imports=skip
+
+.. note::
+   You'll need to use either the ``--follow-imports=skip`` or the
+   ``--follow-imports=error`` option with dmypy.
```

Comment:
Explain why -- it's a bit confusing. Also if it's given in mypy.ini that's enough right?